src/Kmd.Logic.Gateway.Automation/PreValidation/PreValidationBase.cs

```diff
@@ -0,0 +1,171 @@
+﻿using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+
+namespace Kmd.Logic.Gateway.Automation
+{
+    public class PreValidationBase
+    {
+        private readonly MediaConfiguration _mediaConfiguration = new MediaConfiguration()
+        {
+            MaxLogoSizeBytes = 1000000,
+            MaxMarkDownDocumentSizeBytes = 1000000,
+            MaxOpenApiSpecSizeBytes = 20971520,
+            MaxPolicyXmlSizeBytes = 10240,
+            AllowedLogoFileExtensions = new[] { ".png", ".jpeg", ".jpg" },
+            AllowedMarkdownDocumentExtension = ".md",
+            AllowedOpenApiSpecExtension = ".json",
+            AllowedPolicyXmlExtension = ".xml",
+        };
+
+        public PreValidationBase(string folderPath)
+        {
+            this.FolderPath = folderPath;
+            this.ValidationResults = new List<PublishResult>();
+        }
+
+        public string FolderPath { get; }
+
+        public List<PublishResult> ValidationResults { get; }
+
+        protected bool ValidateFile(string path, string enityName, string propName, GatewayFileType fileType)
```

Comment:
I think that `GatewayFileType fileType` is the most important parameter here, so it should be first.

Reply:
moved to first parameter